packages/encryption/src/ec.ts

```diff
@@ -101,6 +101,48 @@ function sharedSecretToKeys(sharedSecret: Buffer): { encryptionKey: Buffer; hmac
   };
 }
 
+/**
+ * @ignore
+ */
+function allHexChars(maybe: string): boolean {
+  return maybe.match(/^[0-9a-f]+$/i) !== null;
+}
+
+/**
+ * @ignore
+ */
+function isValidPublicKey(pub: string): { result: boolean; reason: string | null } {
+  const invalidFormat = {
+    result: false,
+    reason: 'Invalid public key format',
```

Comment:
Maybe it'd be good to use a reason `enum` here, such as

```ts
enum InvalidPublicKeyReason {
  InvalidFormat,
  IsNotPoint
}
```

This'd make it easier to programatically check which one it is. Atm you'd have to compare strings. The text as it is could still be returned, but perhaps as a different property?

Reply:
@kyranjamie Added these changes. Let me know if it's fine or need further improvement.